src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSource.java

```diff
@@ -92,17 +93,23 @@ public ReadsSparkSource(final JavaSparkContext ctx, final ValidationStringency v
 
         setHadoopBAMConfigurationProperties(readFileName, referencePath);
 
+        boolean isBam = IOUtils.isBamFileName(readFileName);
+        if (isBam && intervals != null && !intervals.isEmpty()) {
+            BAMInputFormat.setIntervals(conf, intervals);
+        } else {
+            conf.unset(BAMInputFormat.INTERVALS_PROPERTY);
+        }
+
         rdd2 = ctx.newAPIHadoopFile(
                     readFileName, AnySAMInputFormat.class, LongWritable.class, SAMRecordWritable.class,
                     conf);
 
         return rdd2.map(v1 -> {
             SAMRecord sam = v1._2().get();
-            if (samRecordOverlaps(sam, intervals)) {
-                return (GATKRead)SAMRecordToGATKReadAdapter.headerlessReadAdapter(sam);
+            if (isBam || samRecordOverlaps(sam, intervals)) { // don't check overlaps for BAM since it is done by input format
+                return (GATKRead) SAMRecordToGATKReadAdapter.headerlessReadAdapter(sam);
             }
             return null;
-
         }).filter(v1 -> v1 != null);
     }
 
```

Comment:
There is a TODO below for the `samRecordOverlaps()` method that says "remove this method when https://github.com/broadinstitute/hellbender/issues/559 is fixed", but that ticket has been closed as resolved. Can you create a new ticket "Use `IntervalsSkipList` in `ReadSparkSource.samRecordOverlaps()`, update the docs for the method to point to that ticket, and also add a note to the `samRecordOverlaps()` that it is only used as a fallback option for formats that don't support query-by-interval natively at the Hadoop-BAM layer?